File: tools/browser_back.lua
local BROWSER_SCRIPT = "tools/browser.py"

return {
    name = "browser_back",
    description = "Navigate back to the previous page in browser history.",
    tool = {
        type = "function",
        ["function"] = {
            name = "browser_back",
            description = "Navigate back to the previous page in browser history. Requires browser_navigate to be called first.",
            parameters = {
                type = "object",
                properties = {},
                required = {}
            }
        }
    },
    action = function(input)
        local cmd = "python3 " .. BROWSER_SCRIPT .. " back"
        local p = io.popen(cmd)
        if not p then
            return { success = false, content = "Failed to go back" }
        end
        local result = p:read("*a")
        p:close()
        return { success = true, content = result }
    end
}

File: tools/browser_back.py
"""browser_back tool — go back in browser history."""

name = "browser_back"
description = "Navigate back to the previous page in browser history."
schema = {
    "type": "function",
    "function": {
        "name": "browser_back",
        "description": "Navigate back to the previous page in browser history.",
        "parameters": {"type": "object", "properties": {}, "required": []},
    },
}

_BROWSER_SCRIPT = "tools/browser.py"


def action(input: dict) -> dict:
    import subprocess
    r = subprocess.run(["python3", _BROWSER_SCRIPT, "back"], capture_output=True, text=True, timeout=30)
    return {"success": True, "content": r.stdout or r.stderr}
